gematria/granite/graph_builder_model_inference.cc

```diff
@@ -428,25 +424,31 @@ GraphBuilderModelInference::FromTfLiteModel(
 
   // Get a mapping between the input tensors used in the model and their
   // corresponding indices.
-  auto input_tensor_to_idx =
-      std::make_unique<std::vector<int>>(kNumPossibleInputTensors, -1);
+  std::vector<int> input_tensor_to_idx(kNumPossibleInputTensors, -1);
   for (int idx : (*interpreter)->inputs()) {
     const std::string_view name = (*interpreter)->GetInputName(idx);
-    input_tensor_to_idx->at(static_cast<int>(GetInputTensorFromName(name))) =
-        idx;
+    std::optional<InputTensor> input_tensor = GetInputTensorFromName(name);
+
+    // Report any unexpected tensors.
+    if (!input_tensor.has_value()) {
+      llvm::errs() << "Model contains unexpected tensor `" << name
+                   << "` at index " << idx << '\n';
+      continue;
+    }
+    input_tensor_to_idx.at(static_cast<int>(input_tensor.value())) = idx;
```

Comment:
Why not `input_tensor_to_idx[static_cast<int>(input_tensor.value())] = idx` ?

It's more common (and more efficient, because `at()` always does a range check on the index, but that's not something we need to worry about here).

Same for all the other uses of `std::vector<>::at()`.